src/test/java/uk/gov/companieshouse/api/testdata/service/impl/TestDataServiceImplTest.java

```diff
@@ -919,15 +917,85 @@ void createAcspMembersDataException() throws DataException {
     }
 
     @Test
-    void createAcspMembersDataWithDefaultProfileValues() throws DataException {
+    void createAcspMembersDataWhenProfileIsNotNull() throws DataException {
+        AcspMembersSpec spec = new AcspMembersSpec();
+        spec.setUserId("userId");
+
         AcspProfileData acspProfileData = new AcspProfileData("acspNumber");
-        AcspMembersData expectedMembersData =
-                new AcspMembersData("memberId", "acspNumber", "userId", "active", "role");
+        AcspMembersData acspMembersData = new AcspMembersData("memberId", acspProfileData.getAcspNumber(), "userId", "active", "role");
+        var acspStatus = "active";
+        var acspType = "ltd";
+        var supervisoryBody = "financial-conduct-authority-fca";
+        var membershipDetails = "Membership ID: FCA654321";
 
-        AcspMembersData result = createAcspMembersDataHelper("userId", acspProfileData, expectedMembersData);
-        verifyAcspMembersData(result, "memberId", "acspNumber", "userId", "active", "role");
-        verify(acspProfileService).create(any(AcspProfileSpec.class));
-        verify(acspMembersService).create(any(AcspMembersSpec.class));
+        AcspProfileSpec acspProfile = new AcspProfileSpec();
+        acspProfile.setStatus(acspStatus);
+        acspProfile.setType(acspType);
+        AmlSpec amlSpec = new AmlSpec();
+        amlSpec.setSupervisoryBody(supervisoryBody);
+        amlSpec.setMembershipDetails(membershipDetails);
+
+        acspProfile.setAmlDetails(Collections.singletonList(amlSpec));
+
+        spec.setAcspProfile(acspProfile);
+
+        when(acspProfileService.create(any(AcspProfileSpec.class))).thenReturn(acspProfileData);
+        when(acspMembersService.create(any(AcspMembersSpec.class))).thenReturn(acspMembersData);
+
+        AcspMembersData result = testDataService.createAcspMembersData(spec);
+
+        assertNotNull(result);
+        assertEquals(acspProfileData.getAcspNumber(), result.getAcspNumber());
+        assertEquals(acspMembersData.getAcspMemberId(), result.getAcspMemberId());
+        assertEquals(acspMembersData.getUserId(), result.getUserId());
+        assertEquals(acspMembersData.getStatus(), result.getStatus());
+        assertEquals(acspMembersData.getUserRole(), result.getUserRole());
+
+        verify(acspProfileService).create(argThat(profile ->
+                acspType.equals(profile.getType()) && acspStatus.equals(profile.getStatus())
+                        && profile.getAmlDetails() != null && !profile.getAmlDetails().isEmpty()
+                        && supervisoryBody.equals(
+                                profile.getAmlDetails().getFirst().getSupervisoryBody())
+                        && membershipDetails.equals(
+                                profile.getAmlDetails().getFirst().getMembershipDetails())
+        ));
```

Comment:
This can now be simplified
```suggestion
        verify(acspProfileService).create(acspProfile);
```